Source/Core/Core/HW/WiimoteEmu/SystemBattery.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/HW/WiimoteEmu/SystemBattery.h"
+
+#if defined(_WIN32)
+
+#include <Windows.h>
+
+std::optional<float> GetSystemBatteryLevel()
+{
+  SYSTEM_POWER_STATUS power_status;
+  if (!GetSystemPowerStatus(&power_status))
+  {
+    return std::nullopt;
+  }
+
+  // Not on battery power, treat as 100%
+  if (power_status.ACLineStatus == 1)
+  {
+    return 1.0;
+  }
+
+  if (power_status.BatteryLifePercent != 255)
+  {
+    return (float)power_status.BatteryLifePercent / 100.0;
```

Comment:
You can avoid the cast if you declare the divisor as `100.0f`.

Reply:
Oh right, if one side of the division is a floating-point type, I don't need to cast the other.